crypto.go

```diff
@@ -220,3 +221,19 @@ func preferServerCipherSuites() bool {
 	)
 	return !hasGCMAsm
 }
+
+// Compare two peer IDs by their SHA256 hash.
+// The result will be 0 if H(a) == H(b), -1 if H(a) < H(b), and +1 if H(a) > H(b).
+func comparePeerIDs(p1, p2 peer.ID) int {
+	p1Hash := sha256.Sum256([]byte(p1))
+	p2Hash := sha256.Sum256([]byte(p2))
+	for i := 0; i < sha256.Size; i++ {
+		if p1Hash[i] < p2Hash[i] {
+			return -1
+		}
+		if p1Hash[i] > p2Hash[i] {
+			return 1
+		}
+	}
+	return 0
```

Comment:
We can use `bytes.Compare` for this, no need to do it by hand.

Reply:
Oh cool, the standard library has a function for that. That's convenient.